File: diff-adapter/src/main/java/ru/starksoft/differ/api/OnAdapterRefreshedListener.java
package ru.starksoft.differ.api;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.annotation.UiThread;
import ru.starksoft.differ.adapter.DifferLabels;
import ru.starksoft.differ.adapter.viewmodel.ViewModel;

public interface OnAdapterRefreshedListener {
	@UiThread
	void updateAdapter(@NonNull List<ViewModel> viewModels, @NonNull DifferLabels labels, boolean dontTriggerMoves);

}

File: diff-adapter/src/main/java/ru/starksoft/differ/api/OnAdapterRefreshedListener.kt
package ru.starksoft.differ.api

import androidx.annotation.UiThread
import ru.starksoft.differ.adapter.DifferLabels
import ru.starksoft.differ.adapter.viewmodel.ViewModel

fun interface OnAdapterRefreshedListener {

    @UiThread
    fun updateAdapter(viewModels: List<ViewModel>, labels: DifferLabels, dontTriggerMoves: Boolean)
}